vllm/lora/layers.py

```diff
@@ -326,6 +331,12 @@ def set_lora(
     ):
         self.reset_lora(index)
 
+        if self.lora_config.fully_sharded_loras:
+            tp_rank = get_tensor_model_parallel_rank()
+            shard_size = self.lora_a_stacked.shape[2]
+            start_idx = tp_rank * shard_size
+            lora_a = lora_a[:, start_idx:start_idx + shard_size]
```

Comment:
let's put those `if self.lora_config.fully_sharded_loras:` blocks into a method that is a noop for normal layers and override it in fully sharded implementations